### PR TITLE
[exporterhelper] Add data_type attribute to internal queue metrics

### PR DESCRIPTION
**Pull request by dmitryax**
Add `data_type` attribute to the internal otelcol_exporter_queue_size metric to report the type of data being processed.

All other metrics have the data type reported as part of their names. We could've done the same for queue metrics, but that would introduce a significant breaking change. We want to avoid that until we have all the metrics standardized with OpenTelemetry semantic conventions.

Fixes https://github.com/open-telemetry/opentelemetry-collector/issues/9943

### CODE REVIEW

**Review by songy23 — APPROVED** [review]

File: component/componenttest/obsreporttest.go
Comment (dmitryax) [inline component/componenttest/obsreporttest.go:76]: @open-telemetry/collector-approvers do you think this change needs a breaking change item in API changelog?

Comment (mx-psi) [reply component/componenttest/obsreporttest.go:76]: Yes, IMO we don't need a deprecation period but we should add a changelog entry

**Review by mx-psi — APPROVED** [review]

File: internal/obsreportconfig/obsmetrics/obs_exporter.go
Comment (codeboten) [inline internal/obsreportconfig/obsmetrics/obs_exporter.go:11]: should this be what the [otep](https://github.com/open-telemetry/oteps/pull/259/files#diff-8ab369bae50cfda348c0bd20882db8a1bdbbffb979abc70bff298192a948588bR218) recommends `otel.signal`?

Comment (dmitryax) [reply internal/obsreportconfig/obsmetrics/obs_exporter.go:11]: I'm not sure about it. We use `data_type` as a field for logging. Once the OTEP is ready and we adopt it, the metric is going to change anyway. Keeping an attribute name doesn't help with the migration. Also, `otel.signal` could still change before the OTEP is merged?

File: component/componenttest/obsreporttest.go
Comment (dmitryax) [reply component/componenttest/obsreporttest.go:76]: added the entry

File: internal/obsreportconfig/obsmetrics/obs_exporter.go
Comment (dmitryax) [reply internal/obsreportconfig/obsmetrics/obs_exporter.go:11]: @codeboten I'm open to any attribute name. Let me know what do you think considering my previous comment.

Comment (codeboten) [reply internal/obsreportconfig/obsmetrics/obs_exporter.go:11]: @dmitryax probably ok to go with an attribute consistent w/ logging for now.

**Review by codeboten — APPROVED** [review]
I searched through semantic conventions and couldn't find guidance for attributes that specify signal types. Should an issue be opened there?

**Comment by dmitryax** [comment]
> I searched through semantic conventions and couldn't find guidance for attributes that specify signal types. Should an issue be opened there?

Yeah, that's something that can be done before the OTEP

**Pull request merged**
